auxiliary/crash-handler/main.cpp

```diff
@@ -0,0 +1,54 @@
+#include "recovery.hpp"
+
+#include <handler/handler_main.h>
+#include <tools/tool_support.h>
+
+#if BUILDFLAG(IS_WIN)
+#    include <Windows.h>
+#endif
+
+namespace {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays)
+int actualMain(int argc, char *argv[])
+{
+    // We're tapping into the crash handling by registering a data source.
+    // Our source doesn't actually provide any data, but it records the crash.
+    // Once the crash is recorded, we know that one happened and can attempt to restart
+    // the host application.
+    crashpad::UserStreamDataSources sources;
+    sources.emplace_back(std::make_unique<CrashRecoverer>());
+
+    auto ret = crashpad::HandlerMain(argc, argv, &sources);
+
+    if (ret == 0)
+    {
+        auto *recoverer = dynamic_cast<CrashRecoverer *>(sources.front().get());
+        if (recoverer != nullptr)
+        {
+            recoverer->attemptRecovery();
+        }
+    }
+    return ret;
+}
+
+}  // namespace
+
+// The following is adapted from handler/main.cc
+#if BUILDFLAG(IS_POSIX)
+
+int main(int argc, char *argv[])
+{
+    return actualMain(argc, argv);
```

Comment:
We can `static_assert` to fail the compilation since we explicitly don't support non-Windows platforms